src/Spago/Command/Fetch.purs

```diff
@@ -357,31 +357,67 @@ writeNewLockfile reason allTransitiveDeps = do
 toAllDependencies :: PackageTransitiveDeps -> PackageMap
 toAllDependencies = foldl (Map.unionWith (\l _ -> l)) Map.empty
 
+data CopyOrMove = Copy | Move
+
 getGitPackageInLocalCache :: forall a. PackageName -> GitPackage -> Spago (Git.GitEnv a) Unit
 getGitPackageInLocalCache name package = do
-  let localPackageLocation = Config.getPackageLocation name (GitPackage package)
-  tempDir <- mkTemp' (Just $ printJson Config.gitPackageCodec package)
-  logDebug $ "Cloning repo in " <> tempDir
-  Git.fetchRepo package tempDir >>= case _ of
-    Left err -> die err
-    Right _ -> do
+  let gitCacheLocation = Config.getGlobalGitCacheForRepo package
+      localPackageLocation = Config.getPackageLocation name (GitPackage package)
+  existsGitCache <- FS.exists gitCacheLocation
+  Tuple copyOrMove sourceDir <-
+    if existsGitCache
+      then do
+        shouldCache <- Git.determineShouldCache package.ref gitCacheLocation
+        if shouldCache
+          then
+            logInfo $ "Using global git cache in " <> gitCacheLocation
+          else do
+            fetchRes <- Git.fetchRepo package gitCacheLocation
+            case fetchRes of
+              Left err -> die err
+              Right _ -> pure unit
+        pure (Tuple Copy gitCacheLocation)
+      else do
```

Comment:
Informative comment to keep track of what's happening
```suggestion
      -- Not cached in the global cache, so we download it and then see if it can be cached
      else do
```